src/umfile_utils/perturbIC.py

```diff
@@ -0,0 +1,229 @@
+    shape: tuple or list
+        Shape of the generated perturbation.
+    nullify_poles: bool, optional
+        If set to True, nullifies the perturbation amplitude at the poles.
+
+    Returns
+    ----------
+    pertubation: numpy.ndarray 
+        The generated random perturbation.
+    """
+    perturbation = random_generator.uniform(low = -amplitude, high = amplitude, size = shape)
+    # Set poles to zero (only necessary for ND grids, but doesn't hurt EG)
+    if nullify_poles:
+        perturbation[[0,-1],:] = 0
+    return perturbation
+
+
+def is_field_to_perturb(field, stash_to_perturb):
+    """
+    Check if the field STASH itemcode correspond to the one to perturb.
+    
+    Parameters
+    ----------
+    field : mule.Field
+           Field to check.
+    stash_to_perturb: int
+        STASH itemcode to perturb.
+
+    Returns
+    ----------
+    bool
+        Returns True if the field STASH itemcode corresponds to the one to perturb.
+    """
+    return field.lbuser4 == stash_to_perturb
+
+class AdditionOperator(mule.DataOperator):
+    """
+    Create a mule operator that adds an array to a field, provided that the two have the same shape.
+    
+    Attributes
+    ----------
+    array : numpy.ndarray
+             The array to add to the field.
+    """
+    def __init__(self, array):
+        self.array = array
+
+    def new_field(self, source_field):
+        """
+        Create the new field object by copying the source field.
+        """
+        return source_field.copy()
+
+    def transform(self, source_field, new_field):
+        """
+        Perform the field data manipulation: check that the array and source field data have the same shape and then add them together.
+        """
+        data = source_field.get_data()
+        if (field_shape:=data.shape) != (array_shape:=self.array.shape):
+            raise ValueError(f"Array and field could not be broadcast together with shapes {array_shape} and {field_shape}.")
+        else:
+            return data + self.array
+
+
+def void_validation(*args, **kwargs):
+    """
+    Don't perform the validation, but print a message to inform that validation has been skipped.
+    """
+    print('Skipping mule validation. To enable the validation, run using the "--validate" option.')
+
+
+def main():
+    """
+    Add a bi-dimensional random perturbation to the potential temperature field 'Theta' (STASH itemcode = 4) of a UM fields file.
+    """
+
+    # Define all the variables  
+    STASH_THETA = 4
+
+    # Parse the command line arguments
+    args = parse_args()
+
+    # Create the output filename
+    output_file = create_default_outname(args.ifile) if args.output_path is None else args.output_path
+
+    # Create the random generator.
+    random_generator = create_random_generator(args.seed)
+
+    # Skip mule validation if the "--validate" option is provided
+    if not args.validate:
+        mule.DumpFile.validate = void_validation
```

Comment:
```suggestion
    if not args.validate:
        ff.validate = void_validation
```
Move lines to 210.

After reviewing the subset function I thought about this. 
`mule.DumpFile.validate = void_validation` sets the `void_validation` function to the `validate` method for the `mule.DumpFile` class, which means it would work for ALL instances of `mule.DumpFile`.
We don't want this, but only avoid the validation for the fieldsfile we're working with (only 1 specific instance of `mule.DumpFile`).

Reply:
I added this commit why is it still showing unresolved change? Everything works with the new change